tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/ScopedBuilder.scala

```diff
@@ -198,6 +219,7 @@ class ScopedBuilder
    * The use of this constructor is discouraged in non-legacy code, as deep-EXCEPT syntax impedes readability, since the
    * types of intermediate objects are obfuscated.
    */
+  @deprecated
   def exceptGeneral(
```

Comment:
Why are we marking these `@deprecated` instead of removing them?

Reply:
I'm surprised that these go through CI without generating compiler warnings (or a `@nowarn` to silence them) 🤔 CC @shonfeder

Reply:
The idea is to wait until we start replacing the old builder with the new builder to see if these methods end up being (un)necessary. If we can port without using them we can delete them then, but I'm not 100% sure we can do without.